Tests/Tactics/Sym.lean

```diff
@@ -17,6 +20,29 @@ namespace GCMGmultV8Program
   simp (config := {ground := true}) only [Option.some.injEq] at h_s0_pc h_run
   sym_n 1
   · sorry
-  · exact (sorry : CheckSPAlignment s0)
   · exact (sorry : read_err s0 = .None)
   done
+
+
+namespace SimpleMemory
+
+def program := def_program [
+    (0x4005b8#64 , 0xb9000fe0#32), -- str w0, [sp, #12]
+  ]
+
+#genStepEqTheorems program
+
+example
+    (sf s0 : ArmState)
+    (h_program : s0.program = program)
+    (h_run : sf = run 1 s0)
+    (h_err : read_err s0 = .None)
+    (h_pc : read_pc s0 = 0x4005b8#64)
+    (h_sp : CheckSPAlignment s0)
+    (h_s0_x0 : r (.GPR 0#5) s0 = x)
+    (h_s0_x31 : r (.GPR 31#5) s0 = y) :
+    read_mem_bytes 4 (y + 12#64) s0 = (x.zeroExtend 32) := by
+  sym_n 1
+  sorry
```

Comment:
@bollu and @alexkeizer: Excellent example to kick-start thinking about to get `simp_mem` and `sym_n` get along great together!